dashboard/app/util_test.go

```diff
@@ -231,9 +246,25 @@ func (c *Ctx) setNoObsoletions() {
 	}
 }
 
+func (c *Ctx) decommissionManager(ns, oldManager, newManager string) {
+	c.transformContext = func(c context.Context) context.Context {
+		newConfig := replaceManagerConfig(c, ns, oldManager, func(cfg ConfigManager) ConfigManager {
+			cfg.Decommissioned = true
+			cfg.DelegatedTo = newManager
+			return cfg
+		})
+		return contextWithConfig(c, newConfig)
+	}
+}
+
 func (c *Ctx) decommission(ns string) {
 	c.transformContext = func(c context.Context) context.Context {
-		return contextWithDecommission(c, ns, true)
+		newConfig := replaceNamespaceConfig(c, ns, func(cfg *Config) *Config {
+			ret := *cfg
```

Comment:
This is duplicated in every callback, also subject for bugs. replaceNamespaceConfig should pass a pointer to a copy to the callback. Then also remove return value of the callback.

Reply:
This is actually the tricky part. I'm not sure we want to implement and support full proper cloning of different config parts. 

Doing a shallow copy in the `replaceNamespaceConfig` instead would IMHO make its semantics much more complicated than so. So I thought it'd be better to let the callback decide how to best transform the object. So far shallow copying looks enough.

If we are to enforce config immutability like you mentioned in the comment below, we could as well do `json.Serialize` before and after the callback to ensure the original object was not modified. This is only needed during tests, so the performance penalty is not something to worry about.

Reply:
> If we are to enforce config immutability like you mentioned in the comment below, we could as well do json.Serialize before and after the callback to ensure the original object was not modified. 

Actually, we don't have to repeat the functionality. If we ensure the global config did not change during the test (see last commit), it already covers situations when we accidentally edited it in the callback here.

Reply:
It's a good idea.

Even if we make semantics simple, but error-prone, it won't help.